egs2/TEMPLATE/asr1/asr.sh

```diff
@@ -1342,38 +1346,37 @@ fi
 
 
 packed_model="${asr_exp}/${asr_exp##*/}_${inference_asr_model%.*}.zip"
-if ! "${skip_upload}"; then
-    if [ ${stage} -le 14 ] && [ ${stop_stage} -ge 14 ]; then
-        log "Stage 14: Pack model: ${packed_model}"
+if [ ${stage} -le 14 ] && [ ${stop_stage} -ge 14 ]; then
+    log "Stage 14: Pack model: ${packed_model}"
 
-        _opts=
-        if "${use_lm}"; then
-            _opts+="--lm_train_config ${lm_exp}/config.yaml "
-            _opts+="--lm_file ${lm_exp}/${inference_lm} "
-            _opts+="--option ${lm_exp}/perplexity_test/ppl "
-            _opts+="--option ${lm_exp}/images "
-        fi
-        if [ "${feats_normalize}" = global_mvn ]; then
-            _opts+="--option ${asr_stats_dir}/train/feats_stats.npz "
-        fi
-        if [ "${token_type}" = bpe ]; then
-            _opts+="--option ${bpemodel} "
-        fi
-        if [ "${nlsyms_txt}" != none ]; then
-            _opts+="--option ${nlsyms_txt} "
-        fi
-        # shellcheck disable=SC2086
-        ${python} -m espnet2.bin.pack asr \
-            --asr_train_config "${asr_exp}"/config.yaml \
-            --asr_model_file "${asr_exp}"/"${inference_asr_model}" \
-            ${_opts} \
-            --option "${asr_exp}"/RESULTS.md \
-            --option "${asr_exp}"/RESULTS.md \
-            --option "${asr_exp}"/images \
-            --outpath "${packed_model}"
+    _opts=
+    if "${use_lm}"; then
+        _opts+="--lm_train_config ${lm_exp}/config.yaml "
+        _opts+="--lm_file ${lm_exp}/${inference_lm} "
+        _opts+="--option ${lm_exp}/perplexity_test/ppl "
+        _opts+="--option ${lm_exp}/images "
     fi
+    if [ "${feats_normalize}" = global_mvn ]; then
+        _opts+="--option ${asr_stats_dir}/train/feats_stats.npz "
+    fi
+    if [ "${token_type}" = bpe ]; then
+        _opts+="--option ${bpemodel} "
+    fi
+    if [ "${nlsyms_txt}" != none ]; then
+        _opts+="--option ${nlsyms_txt} "
+    fi
+    # shellcheck disable=SC2086
+    ${python} -m espnet2.bin.pack asr \
+        --asr_train_config "${asr_exp}"/config.yaml \
+        --asr_model_file "${asr_exp}"/"${inference_asr_model}" \
+        ${_opts} \
+        --option "${asr_exp}"/RESULTS.md \
+        --option "${asr_exp}"/RESULTS.md \
+        --option "${asr_exp}"/images \
+        --outpath "${packed_model}"
+fi
 
-
+if ! "${skip_upload}"; then
```

Comment:
Thanks for the PR.
Just a small suggestion. Would it be better to make the structure in this way?
```bash
upload_target="huggingface"  # default as huggingface
...

if ! "${skip_upload}"; then
    if [ ${stage} -le 14 ] && [ ${stop_stage} -ge 14 ]; then
        # Pack the models
    fi

    if [ ${stage} -le 15 ] && [ ${stop_stage} -ge 15 ]; then
        if [ ${upload_target} = "zenodo" ]; then
            # Upload to Zenodo
        else if [ ${upload_target} = "huggingface" ]; then
            # Upload to HF
        else
            echo "Please choose the upload destination from [zenodo, huggingface]"
        fi
    fi
fi
```

Reply:
I actually like the current @Fhrozen's solution, since we can easily switch zenodo or HF hub based on the stage.
We can also upload both zenodo/HF in the current form.
Also, in the future, we may completely move to HF hub (which I really want to based on the issue a number of people encountered in zenodo), then, we can just remove a zenodo stage.
Either is fine, but this is my preference.